tools/bash/clone_herbert_branch.sh

```diff
@@ -25,20 +25,16 @@ if [ "${herbert_branch}" = "" ]; then
 fi
 
 if [[ -d "${HERBERT_DIR}" ]]; then
-    pushd "${HERBERT_DIR}" || exit 1
-    echo_and_run "git fetch -all --tags" &&
-    echo_and_run "git reset --hard \"${herbert_branch}\""
-    popd || exit 1
-
+    run_in_dir "git fetch --all --tags && git reset --hard \"${herbert_branch}\"" \
+               "${HERBERT_DIR}"
 else
     git_clone_cmd="git clone ${HERBERT_URL} --depth 1"
     git_clone_cmd+=" --branch \"${herbert_branch}\" ${HERBERT_DIR}"
     echo_and_run "${git_clone_cmd}"
 fi
 
-pushd "${HERBERT_DIR}" || exit 1
-echo -e "\nBuilding Herbert revision $(git rev-parse HEAD)..."
-popd || exit 1
+run_in_dir "echo -e \"\nBuilding Herbert revision \$(git rev-parse HEAD)...\"" \
```

Comment:
similarly here, defining the command as a variable would make the `run_in_dir` call clearer

Reply:
This will work, but I don't know if it's any clearer
```bash
print_herb_rev_cmd="echo -e \"\nBuilding Herbert revision"
run_in_dir "${print_herb_rev_cmd} \$(git rev-parse HEAD)...\"" "${HERBERT_DIR}"
```

The `git rev-parse` needs to be executed in the `run_in_dir` function, otherwise it's going to execute it within the Horace repository.

Reply:
The other option is to 
```
echo -e "Build Herbert revision \c" \
  && run_in_dir "$(git rev-parse HEAD)..." "${HERBERT_DIR}"
```